Library/Homebrew/github_runner_matrix.rb

```diff
@@ -53,11 +35,11 @@ def initialize(testing_formulae, deleted_formulae, dependent_matrix:)
     freeze
   end
 
-  sig { returns(T::Array[RunnerSpecHash]) }
+  sig { returns(T::Array[T::Hash[String, T.untyped]]) }
```

Comment:
```suggestion
  sig { returns(T::Array[T.any(LinuxRunnerSpec, MacOsRunnerSpec)]) }
```

Reply:
`MacOSRunnerSpec`

Reply:
I'm reluctant to expose these return types, could I trouble you to elaborate on the motivation for this change? It requires callers such as `determine-test-runners` to have an understanding of `{Linux,MacOS}RunnerSpec` and their internals (`.serialize`), which violates the law of Demeter (at least in spirit).

Reply:
Actually, you can use the `RunnerSpec` alias here instead of doing `T.any`.

For the record, I agree with @dduugg above

> I'm reluctant to expose these return types

I also didn't want to use `T.untyped`, and hence relied on the type alias.

Reply:
Both of these should implement `to_json` instead of us having to duplicate the JSON representation as type aliases. And exposing these types is strictly better than using `T.untyped` in my opinion.

Reply:
To be more specific: I want to have either a Ruby struct or a JSON string. There is no need to have this weird middle-layer consisting of a JSON-compatible Ruby hash which will be converted into a JSON string anyways.